### PR TITLE
bioperl 1.7.8 (new formula)

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/bioperl.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    url :stable
+    regex(/href=.*?>BioPerl-(\d+\.\d+\.\d+)\.t/i)
+  end
+
+  depends_on "cpanminus" => :build
+  depends_on "perl"
+
+  def install
+    ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+    system "cpanm", "--self-contained", "-l", libexec, "DBI" unless OS.mac?
```
Comment (SMillerDev) [inline Formula/bioperl.rb:19]:
```suggestion
    on_linux { system "cpanm", "--self-contained", "-l", libexec, "DBI" }
```

**Review by SMillerDev — COMMENTED** [review]
Almost all Perl formulae have a lot of resource blocks with dependencies, does this not need any extra dependencies?

File: Formula/bioperl.rb (hunk 1 above)
Comment (jonchang) [reply Formula/bioperl.rb:19]: We're moving away from using `on_os` blocks in `def install`.

**Comment by jonchang** [comment]
It uses Dist::Zilla, which I suppose is kind of like a Gemfile and figures out the right locked dependencies.

File: Formula/bioperl.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    system "cpanm", "--self-contained", "-l", libexec, "."
+    bin.env_script_all_files libexec, "PERL5LIB" => ENV["PERL5LIB"]
+    Dir[libexec/"bin/bp_*"].each do |executable|
+      name = File.basename executable
+      (bin/name).write_env_script executable, PERL5LIB: ENV["PERL5LIB"]
```
Comment (carlocab) [inline Formula/bioperl.rb:24]:
```suggestion
    libexec.glob("bin/bp_*") do |executable|
      (bin/executable.basename).write_env_script executable, PERL5LIB: ENV["PERL5LIB"]
```

File: Formula/bioperl.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/bioperl.rb:22]: I thought that was mostly/only for things that can use ternary operators? This is the first PR in core that I've seen use `unless`.

**Pull request closed without merging**